crates/ruff/src/rules/flake8_simplify/rules/ast_unary_op.rs

```diff
@@ -98,8 +98,11 @@ pub(crate) fn negation_with_equal_op(
     }
 
     // Avoid flagging issues in dunder implementations.
-    if let ScopeKind::Function(def) = &checker.semantic_model().scope().kind {
-        if DUNDER_METHODS.contains(&def.name) {
+    if let ScopeKind::Function(ast::StmtFunctionDef { name, .. })
+    | ScopeKind::AsyncFunction(ast::StmtAsyncFunctionDef { name, .. }) =
```

Comment:
Do we need to distinguish between `Function` and `AsyncFunction`. If not, having a single `AnyFunction` kind might be easier to use (it will increase the enum size by 8 bytes because it now requires two discriminates).

Reply:
Leaving this for now, but what do you mean by "two discriminates" here?